src/test/java/com/salesforce/dynamodbv2/UpdateTest.java

```diff
@@ -208,74 +263,164 @@ void updateConditionalFail(TestArgument testArgument) {
     }
 
     @ParameterizedTest
     @ArgumentsSource(DefaultArgumentProvider.class)
-    void updateHkRkTableWithGsi(TestArgument testArgument) {
-        runUpdateHkRkTableWithGsiTest(testArgument, TABLE3);
+    void updateGsiFieldAndQueryGsi(TestArgument testArgument) {
+        testArgument.forEachOrgContext(org -> {
+            String tableName = TABLE3;
+
+            QueryRequest queryGsiRequest = new QueryRequest().withTableName(tableName)
+                .withKeyConditionExpression("#name = :value")
+                .withExpressionAttributeNames(ImmutableMap.of("#name", GSI_HK_FIELD))
+                .withExpressionAttributeValues(ImmutableMap.of(":value",
+                    createStringAttribute(GSI_HK_FIELD_VALUE)))
+                .withIndexName("testGsi");
+            assertEquals(1, testArgument.getAmazonDynamoDb().query(queryGsiRequest).getItems().size());
+
+            Map<String, AttributeValue> updateItemKey = ItemBuilder
+                .builder(testArgument.getHashKeyAttrType(), HASH_KEY_VALUE)
+                .rangeKey(S, RANGE_KEY_S_VALUE)
+                .build();
+            UpdateItemRequest updateItemRequest = new UpdateItemRequest()
+                .withTableName(tableName)
+                .withKey(updateItemKey)
+                .withUpdateExpression("set #gsiHkField = :gsiHkValue")
+                .withExpressionAttributeNames(ImmutableMap.of("#gsiHkField", GSI_HK_FIELD))
+                .withExpressionAttributeValues(ImmutableMap.of(
+                    ":gsiHkValue", createStringAttribute(GSI_HK_FIELD_VALUE)));
+            testArgument.getAmazonDynamoDb().updateItem(updateItemRequest);
+
+            assertEquals(
+                ItemBuilder.builder(testArgument.getHashKeyAttrType(), HASH_KEY_VALUE)
+                    .someField(S, SOME_FIELD_VALUE + tableName + org)
+                    .rangeKey(S, RANGE_KEY_S_VALUE)
+                    .gsiHkField(S, GSI_HK_FIELD_VALUE)
+                    .build(),
+                getItem(testArgument.getAmazonDynamoDb(),
+                    tableName,
+                    HASH_KEY_VALUE,
+                    testArgument.getHashKeyAttrType(),
+                    Optional.of(RANGE_KEY_S_VALUE)));
+            assertEquals(new HashMap<>(updateItemKey), updateItemRequest.getKey()); // assert no side effects
+            assertEquals(tableName, updateItemRequest.getTableName()); // assert no side effects
+
+            // query the record based on index
+            assertEquals(2, testArgument.getAmazonDynamoDb().query(queryGsiRequest).getItems().size());
+        });
     }
 
     @ParameterizedTest
     @ArgumentsSource(DefaultArgumentProvider.class)
-    void updateHkRkTableWithPkFieldInGsi(TestArgument testArgument) {
-        runUpdateHkRkTableWithGsiTest(testArgument, TABLE5);
-    }
-
-    private void runUpdateHkRkTableWithGsiTest(TestArgument testArgument, String tableName) {
+    void updateFieldInMultipleGsis(TestArgument testArgument) {
```

Comment:
New test case where we update a field that is used in multiple GSIs and requires us to fan out. At the end we query both indexes to make sure both queries work.